### PR TITLE
Pybind11 integration

### PR DESCRIPTION
**Pull request by mingruimingrui**
# Context
When using the python package and the c library throws an exception, usually this happens
```python
Segmentation fault (core dumped)
```

This happens because the c exception was not recognized by python and thus it could only throw a very generic exception to halt the program.

# What is this PR for?
This PR addresses the issue above by using [Pybind11](https://github.com/pybind/pybind11) to create a python compliant c++ interface for the python library to utilize. This integration script can be found in [`src/py_opencc.py`](https://github.com/mingruimingrui/OpenCC/blob/990ec746f347b84ffdb4b7bfe92ee7e80b8fe144/src/py_opencc.cpp)

By doing so the following features are achieved
- Errors/Exception that the c library throws, will now be correctly captured on the python side
- C++ type checking will now be correctly done when calling the associated python functions.
- A minor improvement in performance is achieved since c objects can now be directly converted into python objects.

# Additional details
Since building with Pybind11 requires python development headers, when distributing the python package on PyPI, a distribution has to be built for each minor version of Python. (eg. 2.7, 3.5, 3.6, 3.7, 3.8)

To help with this chore, the [`bdist_array.sh`](https://github.com/mingruimingrui/OpenCC/blob/feature/experimental-pybind/bdist_array.sh) and [`bdist_array.cmd`](https://github.com/mingruimingrui/OpenCC/blob/feature/experimental-pybind/bdist_array.cmd) scripts are added.
When called, they would build the `.whl` files for all specified python versions and store them into the `./dist/` directory.

> Take note that both scripts require [`conda`](https://docs.conda.io/en/latest/miniconda.html).
Also it probably is better practice to build with docker instead.

Another smaller but still significant change is that `setup.py` is moved from `./python/setup.py` to `./setup.py`. This change was made so that the source code can now be uploaded to PyPI for users using systems without official binaries (though they would have to install the build tools beforehand).

### CODE REVIEW

File: src/py_opencc.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+class PYOpenCC {
+private:
+    opencc_t converter;
+
+public:
+    std::string config;
+
+    PYOpenCC(const char* configFileName) {
+        config = configFileName;
+        converter = opencc_open(configFileName);
```
Comment (BYVoid) [inline src/py_opencc.cpp:13]: Why don't we depend on the C++ SimpleConverter API?

File: .gitignore [hunk 2]
```diff
@@ -12,3 +12,5 @@
 /test/dict.bin
 /xcode
 /node_modules
+/dist
```
Comment (BYVoid) [inline .gitignore:15]: Can you please also update `.npmignore`?

Comment (mingruimingrui) [reply .gitignore]: Done fdab98a

File: src/py_opencc.cpp (hunk 1 above)
Comment (mingruimingrui) [reply src/py_opencc.cpp]: Good point! Saves rewriting of code.

**Comment by BYVoid** [comment]
LGTM

File: src/py_opencc.cpp (hunk 1 above)
Comment (mingruimingrui) [reply src/py_opencc.cpp]: Done 32a1ec7 f52dad0

**Pull request merged**